File: server/computed_functions/01_app_metadata.compute.mjs
// 01_app_metadata — STARTUP. Returns { name, version, started_at }. Not referenced by
// server.dev.yaml directly but useful for SDK examples and future {{fn:app_metadata.X}} use.
const startedAt = new Date().toISOString();

export default function compute(_ctx, _path) {
  return {
    name: process.env.APP_NAME ?? "demo",
    version: process.env.APP_VERSION ?? "0.0.0",
    started_at: startedAt,
  };
}

export const scope = "STARTUP";

File: server/computed_functions/01_app_metadata.compute.py
"""01_app_metadata — STARTUP. Returns {name, version, started_at}."""
from __future__ import annotations
import os
from datetime import datetime, timezone

_started_at = datetime.now(timezone.utc).isoformat()


def compute(_ctx, _path=None):
    return {
        "name": os.environ.get("APP_NAME", "demo"),
        "version": os.environ.get("APP_VERSION", "0.0.0"),
        "started_at": _started_at,
    }


scope = "STARTUP"
